src/App.jsx

```diff
@@ -1,66 +1,80 @@
-import React, { useEffect } from 'react';
-import { Routes, Route, useNavigate, useLocation } from 'react-router-dom';
+import React from 'react';
+import { Routes, Route, Navigate, useNavigate } from 'react-router-dom';
 
 import Header from './components/Header/Header';
 import Registration from './components/Registration/Registration';
 import Login from './components/Login/Login';
 import Courses from './components/Courses/Courses';
-import CreateCourse from './components/CreateCourse/CreateCourse';
+import CourseForm from './components/CourseForm/CourseForm';
 import CourseInfo from './components/CourseInfo/CourseInfo';
+import PrivateRoute from './routes/helpers/PrivateRoute';
+import { ROUTES } from './routes/routes';
+import { ADMIN } from './roles/roles';
 
 function App() {
 	const navigate = useNavigate();
-	const { pathname } = useLocation();
-
-	// Redirecting to the login page if the user is not authorized
-	useEffect(() => {
-		let userToken = localStorage.getItem('user');
-		if (!userToken && pathname !== '/login' && pathname !== '/registration') {
-			navigate('/login');
-		}
-		if (pathname === '/') {
-			if (!userToken) {
-				navigate('/login');
-			} else {
-				navigate('/courses');
-			}
-		}
-	}, [navigate, pathname]);
 
 	/**
 	 * Navigate to the Courses page
 	 *
 	 * @param {boolean} courseCreatedSuccess
 	 */
-	let navigateToCourses = (courseCreatedSuccess) => {
+	const navigateToCourses = (courseCreatedSuccess) => {
 		if (!courseCreatedSuccess) return;
 
 		// Navigate to the Courses list
-		navigate('/courses');
+		navigate(ROUTES.COURSES);
 	};
 
 	return (
-		<main>
+		<main className='mb-5'>
 			<div className='container'>
 				<Header />
 				<Routes>
-					<Route path='/registration' element={<Registration />} />
-					<Route path='/login' element={<Login />} />
+					<Route path={ROUTES.REGISTRATION} element={<Registration />} />
+					<Route path={ROUTES.LOGIN} element={<Login />} />
+					<Route
+						path={ROUTES.HOME}
+						element={
+							<PrivateRoute>
+								<Navigate to={ROUTES.COURSES} />
+							</PrivateRoute>
+						}
+					/>
+					<Route
```

Comment:
It should be PrivateRoute instead of Route. Like
					<PrivateRoute roles={[ADMIN]}
						path={ROUTES.COURSE_ADD}
						element={
								<CourseForm onCreateCourseButtonClick={navigateToCourses} />
						}
					/>

Reply:
It's a Private Route implementation in react-router v6.
Example: https://dev.to/iamandrewluca/private-route-in-react-router-v6-lg5
Example: https://www.robinwieruch.de/react-router-private-routes/